bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/handler/EcobeeUtils.java

```diff
@@ -87,9 +89,12 @@ public static State undefOrPoint(@Nullable String value) {
         return value == null ? UnDefType.UNDEF : new PointType(value);
     }
 
-    public static State undefOrDate(@Nullable Date date, TimeZoneProvider timeZoneProvider) {
-        return date == null ? UnDefType.UNDEF
-                : new DateTimeType(ZonedDateTime.ofInstant(date.toInstant(), timeZoneProvider.getTimeZone()));
+    public static State undefOrDate(@Nullable Instant instant) {
+        return instant == null ? UnDefType.UNDEF : new DateTimeType(ZonedDateTime.ofInstant(instant, ZoneId.of("UTC")));
```

Comment:
This overload should also use `TimeZoneProvider`, i.e.:
```suggestion
    public static State undefOrDate(@Nullable Instant instant, TimeZoneProvider timeZoneProvider) {
        return instant == null ? UnDefType.UNDEF : new DateTimeType(instant.atZone(timeZoneProvider.getTimeZone()));
```

This is an unfortunate side-effect of the way `DateTimeType` currently works - see openhab/openhab-core#2898. Without providing the time-zone, raw formatting will be in UTC.

Reply:
Now I'm confused. Don't I want the time zone to be UTC, because that's what it is in the json returned from Ecobee. That way when it gets displayed in openHAB it will be shown in the user's local time.

Reply:
Here is `lastModified` in the console.
```
E1_Info_LastModified (Type=DateTimeItem, State=2023-01-08T12:25:28.000+0000, Label=Last Modified, Category=time, Groups=[gMain_Info])
```
I made a change to the thermostat that caused this item to change at 7:25 AM US EST (UTC-5). So I believe it is correct.

Reply:
I'm sorry for the confusion, and indeed it is kind of a mess until we get that `DateTimeType` issue sorted out. No matter if you provide it in UTC or local time-zone, it will be the same exact moment in time, and with the proper state description, it will be shown as correct local datetime in both cases.

The difference occurs when you omit the state description. In this case the raw `ZonedDateTime` will be shown, and this will of course be different. Here it's preferable to use the configured time-zone, not UTC, as the user will expect to see local time. When the `ZonedDateTime` is constructed with the local time-zone, the raw formatting will also show the time in local time-zone.

So in other words, I'm simply requesting a conversion from UTC to local time-zone for the provided `ZonedDateTime`. In essence it will then be exactly the same as the result of the `undefOrDate` `LocalDateTime` overload. Of course you could argue that all DTO's could have used `Instant` then, but your code looks very clean to me, and you also have the advantage of dealing with the UTC vs. local time logic in your handler rather than deep inside some DTO or deserializer.

Reply:
Yes, I see now. I was looking at the date in the UI formatted with a state description, which was correct. I see now that if I looked at the raw date without the state description it would be in UTC time.

Thanks for that explanation.

Reply:
Ok, I think I might finally have it. 🤞

Reply:
It looks good!